### ISSUE TITLE
0 failed to be found in array

### ISSUE THREAD

**Issue by vors** [pester/Pester#256]
``` powershell
Describe foo {
    It "should find 0 in range" {
        0 | Should be @(0..2)
    }

    It "should find 1 in range" {
        1 | Should be @(0..2)
    }

    It "should find 2 in range" {
        2 | Should be @(0..2)
    }
}
```

![image](https://cloud.githubusercontent.com/assets/816680/5798134/531834b6-9f79-11e4-9cb8-eac2a9e776b4.png)

**Comment by dlwyatt** [comment]
That's interesting, but we can't fix this and #246 at the time time; this `$scalar | Should Be $array` behavior is always going to fail  once #246 is done.

**Comment by vors** [comment]
Yeah, that's fine. The reason why I use such syntax now:
https://github.com/SublimeText/PowerShell/blob/dev/tests/pester/Syntax.Tests.ps1#L65

Maybe it make sense to add syntax for range comparison, like
```powershell
$errorCount | Should be in (0..2)
```

Of course, I can use something like
```powershell
(0..2) -contains $errorCount | Should be $true
```
but then I loose context on failure.

**Comment by dlwyatt** [comment]
In the short term, here's a goofy workaround:

```posh
0 | Should Be @(0;0..2)
```

By sticking an extra 0 element into the array, it causes the `$array -eq $scalar` expression to return a 2-element array instead of a single-element array.  This evaluates to $true no matter what's in the array (where a single-element array's boolean evaluation depends on what's in the array; 0 is $false.)

**Comment by dlwyatt** [comment]
>Of course, I can use something like
>    (0..2) -contains $errorCount | Should be $true
>but then I loose context on failure.

Just curious:  with the changes that are getting added in #266 , would you still want to see a "BeIn" operator, or would you use the `Should Be $true` option with your own logic?

**Comment by vors** [comment]
I think with #266 there is no need in `BeIn`. This bug is just an observation of existing inconsistency. With fix for #246 it should gone.
